documents/sim-objects/branch-predictors.md

```diff
@@ -0,0 +1,76 @@
+# Branch Predictors
+
+## Branch Predictors in gem5
+1. 2-bit local
+2. Bi-Mode
+3. Tournament
+4. Loop predictor
+5. TAGE
+6. Statistical Corrector
+7. Multi Perspective Perceptron predictor
+
+### Local Predictor
+The local predictor is a 1 level branch predictor that has a history table in which each entry is a n bit saturating counter. The history table is indexed by the PC, each counter in history table has 2^N states, if it predicts taken, branch target table in the indirector predictor is used obtain the branch address.  
+
+### Bi-mode Predictor
+The bi-mode predictor is a two-level branch predictor that has three seprate history arrays: a taken array, a not-taken array, and a choice array. The taken/not-taken arrays are indexed by a hash of the PC and the global history. The choice array is indexed by the PC only. Because the taken/not-taken arrays use the same index, they must be the same size.
+
+The bi-mode branch predictor aims to eliminate the destructive aliasing that occurs when two branches of opposite biases share the same global history pattern. By separating the predictors into taken/not-taken arrays, and using the branch's PC to choose between the two, destructive aliasing is reduced.
```

Comment:
I would suggest to have an example here, but is it too trivial to the target readers?